sunpy/net/jsoc/jsoc.py

```diff
@@ -38,9 +38,14 @@ def __init__(self, table=None):
         table : `astropy.table.Table`
         """
 
-        self.table = table
-        self.query_args = None
-        self.requests = None
+        if isinstance(table, list) and isinstance(table[0], type(self)):
+            self.table = table[0].table
+            self.query_args = table[0].query_args
+            self.requests = table[0].requests
+        else:
+            self.table = table
+            self.query_args = table.query_args if hasattr(table, 'query_args') else None
+            self.requests = table.requests if hasattr(table, 'requests') else None
```

Comment:
You can refactor this logic a little to make it so that it's fewer lines.

```suggestion
        if isinstance(table, list) and isinstance(table[0], type(self)):
            table = table[0].table
        self.table = table
        self.query_args = table.query_args if hasattr(table, 'query_args') else None
        self.requests = table.requests if hasattr(table, 'requests') else None
```